clang/lib/Sema/SemaInit.cpp

```diff
@@ -6496,6 +6496,17 @@ static bool canPerformArrayCopy(const InitializedEntity &Entity) {
   return false;
 }
 
+static const FieldDecl *GetConstField(const RecordDecl *RD) {
```

Comment:
```suggestion
static const FieldDecl *getConstField(const RecordDecl *RD) {
```

Also, does this need to consider base types?

Reply:
C has no base classes, so nope!

Reply:
Ah, I see!  Would love an assert to that effect.

Reply:
No assertion is possible, `CXXRecordDecl` has bases, `RecordDecl` does not. I mean, I could use `dyn_cast` and go that route, but then the assertion gets awkward because it requires multiple expressions.

Reply:
Ah, just `assert(!LangOpts.CPlusPlus)` would be enough, but just `assert(!isa<CXXRecordDecl>(RD))` is as well (since all recorddecls in C++ mode are CXXRecordDecl)

Reply:
Ah fair enough, I've added an assert